src/Magnum/Audio/Buffer.cpp

```diff
@@ -0,0 +1,54 @@
+/*
+    This file is part of Magnum.
+
+    Copyright © 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019
+              Vladimír Vondruš <mosra@centrum.cz>
+    Copyright © 2019 Guillaume Jacquemin <williamjcm@users.noreply.github.com>
+
+    Permission is hereby granted, free of charge, to any person obtaining a
+    copy of this software and associated documentation files (the "Software"),
+    to deal in the Software without restriction, including without limitation
+    the rights to use, copy, modify, merge, publish, distribute, sublicense,
+    and/or sell copies of the Software, and to permit persons to whom the
+    Software is furnished to do so, subject to the following conditions:
+
+    The above copyright notice and this permission notice shall be included
+    in all copies or substantial portions of the Software.
+
+    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+    DEALINGS IN THE SOFTWARE.
+*/
+
+#include "Buffer.h"
+
+#include "Magnum/Math/Functions.h"
+
+namespace Magnum { namespace Audio {
+
+Buffer& Buffer::setLoopPoints(Int loopStart, Int loopEnd) {
+    ALint sizeInBytes;
+    ALint channels;
+    ALint bits;
+
+    alGetBufferi(_id, AL_SIZE, &sizeInBytes);
+    alGetBufferi(_id, AL_CHANNELS, &channels);
+    alGetBufferi(_id, AL_BITS, &bits);
+
+    ALint lengthInSamples = sizeInBytes * 8 / (channels * bits);
+
+    ALint loopPoints[2] = {
+        Math::max<ALint>(0, loopStart),
+        Math::min<ALint>(loopEnd, lengthInSamples)
+    };
```

Comment:
What happens when `loopStart` > `loopEnd`? Does the extension say anything about that?

Reply:
> An AL_INVALID_VALUE error is generated if alBufferiv is called with offset
> values for AL_LOOP_POINTS_SOFT where the second value is not greater than
> the first value.

Right.

Reply:
Ok, so (similarly to GL) this is user's responsibility to check, I guess.

Reply:
Yeah.